plugins/src/hashlookup/select.go

```diff
@@ -0,0 +1,241 @@
+
+	var from string
+	var to string
+
+	// Prepare a 'From' value
+	switch expr := rangeCond.From.(type) {
+	case *sqlparser.SQLVal:
+		switch expr.Type {
+		case sqlparser.IntVal, sqlparser.FloatVal, sqlparser.StrVal:
+			from = string(expr.Val)
+		default:
+			return nil, fmt.Errorf("Invalid BETWEEN 'from' value: %v (type %v)", string(expr.Val), expr.Type)
+		}
+	default:
+		return nil, fmt.Errorf("Invalid BETWEEN 'from' value: %v", strings.Trim(sqlparser.String(rangeCond.From), "'"))
+	}
+
+	// Prepare a 'To' value
+	switch expr := rangeCond.To.(type) {
+	case *sqlparser.SQLVal:
+		switch expr.Type {
+		case sqlparser.IntVal, sqlparser.FloatVal, sqlparser.StrVal:
+			to = string(expr.Val)
+		default:
+			return nil, fmt.Errorf("Invalid BETWEEN 'to' value: %v (type %v)", string(expr.Val), expr.Type)
+		}
+	default:
+		return nil, fmt.Errorf("Invalid BETWEEN 'to' value: %v", strings.Trim(sqlparser.String(rangeCond.To), "'"))
+	}
+
+	// Build resulting fields list
+	var fields [][2]string
+	fields = append(fields, [2]string{colNameStr + "_from", from})
+	fields = append(fields, [2]string{colNameStr + "_to", to})
+
+	return fields, nil
+}
+
+func handleSelectWhereParenExpr(expr *sqlparser.Expr, topLevel bool, parent *sqlparser.Expr) ([][2]string, error) {
+	parentBoolExpr := (*expr).(*sqlparser.ParenExpr)
+	boolExpr := parentBoolExpr.Expr
+
+	// If parent is the top level, bool must is needed
+	var isThisTopLevel = false
+	if topLevel {
+		isThisTopLevel = true
+	}
+
+	return handleSelectWhere(&boolExpr, isThisTopLevel, parent)
+}
+
+func buildNestedFuncStrValue(nestedFunc *sqlparser.FuncExpr) (string, error) {
+	return "", errors.New("Unsupported function: " + nestedFunc.Name.String())
+}
+
+func buildComparisonExprRightStr(expr sqlparser.Expr) (interface{}, error) {
+	var rightStr string
+	var err error
+
+	switch expr := expr.(type) {
+	case *sqlparser.SQLVal:
+		// Use string value type only
+		rightStr = sqlparser.String(expr)
+		rightStr = strings.Trim(rightStr, "'")
+
+	case *sqlparser.BoolVal, sqlparser.BoolVal:
+		rightStr = sqlparser.String(expr)
+
+	case *sqlparser.GroupConcatExpr:
+		return nil, errors.New("group_concat not supported")
+
+	case *sqlparser.FuncExpr:
+		// Parse nested
+		//funcExpr := expr.(*sqlparser.FuncExpr)
+		//rightStr, err = buildNestedFuncStrValue(funcExpr)
+		rightStr, err = buildNestedFuncStrValue(expr)
+		if err != nil {
+			return nil, err
+		}
+
+	case *sqlparser.ColName:
+		if sqlparser.String(expr) == "exist" {
+			return nil, errors.New("'exist' expression currently not supported")
+		}
+		return nil, errors.New("Column name on the right side of compare operator is not supported")
+
+	case sqlparser.ValTuple:
+		rightStr = sqlparser.String(expr)
+
+	default:
+		return nil, fmt.Errorf("Unexpected SQL expression right part's type: %T", expr)
+	}
+
+	return rightStr, err
+}
+
+func handleSelectWhere(expr *sqlparser.Expr, topLevel bool, parent *sqlparser.Expr) ([][2]string, error) {
```

Comment:
/*
 * Handle WHERE statement.
 *
 * Receives:
 * expr     - SQL expression to process
 * topLevel - whether it's a top level expression
 * parent   - container of the expression
 */